generators/generator-base-private.js

```diff
@@ -1463,6 +1463,34 @@ module.exports = class extends Generator {
         return primaryKeyType;
     }
 
+    /**
+     * Returns the primary key value based on authentication type, DB, given association and default value
+     *
+     * @param {string} authenticationType - the auth type
+     * @param {string} databaseType - the database type
+     * @param {any} relationships - relationships
+     * @param {string} defaultValue - default value
+     */
+    getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, defaultValue) {
```

Comment:
An idea:
  - the function has way too many arguments, and most of them (`authenticationType` and `relationships`) are only used to get the PK type
  - the PK type can be passed to this function as arg
  - the result `getBlablabla(primaryKeyType, databaseType, defaultValue)` is better, but still: the name doesn't fit
  - `getPrimaryKeyValue(...)` is an improvement over the previous version
  - you can further refactor this, but as it's only used once or twice... there may not be any need for breaking the switch-case into something like getPrimaryKeyValueForStringKey, or ForUUIDKey, etc. (you may have to keep this in mind if you plan to refactor more than one file though)

By the way, **where are the tests?**